src/modules/roc_sndio/target_sox/roc_sndio/sox_sink.cpp

```diff
@@ -134,6 +139,11 @@ void SoxSink::write(audio::Frame& frame) {
 }
 
 bool SoxSink::prepare_() {
+    if (buffer_size_ == 0) {
```

Comment:
I think this check is incorrect:

```
if (buffer_size_ == 0) {
```

With this check, we update `buffer_size_` only if it was zero, i.e. only if `config.sample_rate` was zero in constructor. However, even if `sample_rate` was NOT zero in constructor, it may change after opening device (depending on what is supported by device).

So I think we should recompute `buffer_size_` unconditionally.

Reply:
done.